src/test/agent/test_instance.py

```diff
@@ -0,0 +1,62 @@
+import unittest
+from mock import patch, MagicMock
+from spacel.agent.instance import InstanceManager
+from spacel.model import AgentManifest
+
+
+class TestInstanceManager(unittest.TestCase):
+    def setUp(self):
+        self.instance = InstanceManager()
+        self.manifest = AgentManifest({
+            'healthcheck': {
+                'endpoint': 'test',
+                'virtual_host': 'spacel',
+                'protocol': 'https'
+            }
+        })
+
+    @patch('spacel.agent.instance.InstanceManager.check_health')
+    def test_health(self, mock_check_health):
+        mock_check_health.return_value = True
+        result = self.instance.health(self.manifest, 1)
+        self.assertEqual(result, True)
+
+    @patch('spacel.agent.instance.InstanceManager.check_health')
+    def test_health_fail(self, mock_check_health):
+        mock_check_health.return_value = False
+        result = self.instance.health(self.manifest, 1)
```

Comment:
Nit: all these 1 second timeouts will drag on; can we tune `backoff_scale` and aim for <100ms per test?